### ISSUE TITLE
SQL query with wrong parameter value crashes database

### ISSUE THREAD

**Issue by damienhollis** [cockroachdb/cockroach#30538]
**BUG REPORT**

Please describe the issue you observed, and any steps we can take to reproduce it:

- Which version of CockroachDB are you using?
v2.1.0-beta.20180917

- What did you do?
Executed a SQL query via JDBC where one of the parameters should have been a UUID but was supplied truncated (i.e. e81bb788-2291-4b6e-9cf3-b237fe6c2f3 should have been e81bb788-2291-4b6e-9cf3-b237fe6c2f30).  This caused the database to panic.  

The column that was being queried is declared as a UUID in the table definition. 

- What did you expect to see?
An error back from cockroach that the parameter was the wrong type, it should definitely not have caused cockroachdb to panic.

- What did you see instead?
```
cockroachdb_1       | * ERROR: [n1,client=172.24.0.1:55762,user=root] a panic has occurred!
cockroachdb_1       | *
cockroachdb_1       | panic while executing 1 statements: SELECT _, _ FROM _ WHERE _ = $1; caused by could not parse "e81bb788-2291-4b6e-9cf3-b237fe6c2f3" as type uuid: uuid: incorrect UUID length: e81bb788-2291-4b6e-9cf3-b237fe6c2f3
cockroachdb_1       | 
cockroachdb_1       | goroutine 303 [running]:
cockroachdb_1       | runtime/debug.Stack(0x2f940c0, 0xc4264d7fc0, 0x3)
cockroachdb_1       | 	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/util/log.ReportPanic(0x2f940c0, 0xc4264d7fc0, 0xc42025b980, 0x28d8b20, 0xc42705f9e0, 0x1)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:213 +0xa5
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc426478a00, 0x2f940c0, 0xc4264d7fc0, 0x2902500, 0xc426de5980)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:668 +0x2df
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc426478a00, 0x2f940c0, 0xc4264d7fc0)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:414 +0x61
cockroachdb_1       | panic(0x2902500, 0xc426de5980)
cockroachdb_1       | 	/usr/local/go/src/runtime/panic.go:502 +0x229
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).assignPlaceholders(0xc426479978, 0xc400000003, 0x2)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:11718 +0x51e9
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).assignPlaceholders(0xc426479978, 0x4, 0x18)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:11443 +0x2e71
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).assignPlaceholders(0xc426479978, 0x5, 0x8)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:11416 +0x2552
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).assignPlaceholders(0xc426479978, 0x9, 0x7f6eb08065a8)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:11179 +0x45f
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).assignPlaceholders(0xc426479978, 0xa, 0x0)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:11183 +0x3a5
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/opt/norm.(*Factory).AssignPlaceholders(0xc426479978)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.go:167 +0x3b
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc426478e38, 0x2f94180, 0xc42705f5c0, 0x2f97800, 0xc4267e36c0, 0xc4273d8730, 0x2, 0x2, 0xc427060380, 0x3c, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:465 +0x645
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*planner).optionallyUseOptimizer(0xc426478e38, 0x2f94180, 0xc42705f5c0, 0xc4266ac2c0, 0x16, 0xc4266ef017, 0xe, 0x0, 0x1, 0x0, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:534 +0xce
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc426478a00, 0x2f94180, 0xc42705f5c0, 0x2f97800, 0xc4267e36c0, 0xc4273d8730, 0x2, 0x2, 0xc427060380, 0x3c, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:753 +0x182
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc426478a00, 0x2f94180, 0xc42705f5c0, 0x2f97800, 0xc4267e36c0, 0xc4273d8730, 0x2, 0x2, 0xc427060380, 0x3c, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:403 +0xb42
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc426478a00, 0x2f94180, 0xc42705f5c0, 0x2f97800, 0xc4267e36c0, 0xc4273d8730, 0x2, 0x2, 0xc427060380, 0x3c, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x341
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc426478a00, 0x2f940c0, 0xc4264d7fc0, 0xc426502c50, 0x0, 0x0)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1136 +0x13fe
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc4204f1b00, 0x2f940c0, 0xc4264d7fc0, 0xc4266ef017, 0xe, 0xc4266ef009, 0x4, 0x0, 0x0, 0x2f75680, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1bb
cockroachdb_1       | github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3(0xc4204f1b00, 0x2f940c0, 0xc4264d7fc0, 0xc4266a81c0, 0x5400, 0x15000, 0xc4208b2cc0, 0xc426502c50, 0xc426502c40, 0xc4263b8cd0, ...)
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x122
cockroachdb_1       | created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl
cockroachdb_1       | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0xf04
cockroachdb_1       | 
cockroachdb_1       | *
cockroachdb_1       | * ERROR: [n1,client=172.24.0.1:55762,user=root] Reported as error bd6c5b11700747b4bf45d35a21832a44
cockroachdb_1       | *
```
- What was the impact?
The cockroach node crashed.

**Comment by andy-kimball** [comment]
Hi @damienhollis, I've spent a while trying to repro this, but cannot. Can you provide snippets of your client Java code so I can see how you're preparing and executing?

**Comment by andy-kimball** [comment]
Just after adding that last comment, I stumbled upon a repro. So no need to give more details. Thanks for reporting the issue.

**Comment by andy-kimball** [comment]
Fixed by pull request #30860.
